sonic_installer/main.py

```diff
@@ -591,7 +591,18 @@ def install(url, force, skip_platform_check=False, skip_migration=False, skip_pa
 
         echo_and_log("Installing image {} and setting it as default...".format(binary_image_version))
         with SWAPAllocator(not skip_setup_swap, swap_mem_size, total_mem_threshold, available_mem_threshold):
-            bootloader.install_image(image_path)
+            try:
+                bootloader.install_image(image_path)
+            except BaseException as e:
+                # When install image failed with exception, image is partial installed
+                # Fully installed image will update SONiC environment by update_sonic_environment
+                # For partial installed image, only need delete image folder
+                echo_and_log('Image install failed with exception: {}'.format(e))
+                echo_and_log('Delete partial installed image: {}'.format(binary_image_version))
+                new_image_dir = bootloader.get_image_path(binary_image_version)
+                run_command(["rm", "-rf", new_image_dir])
```

Comment:
>run_command [](http://example.com/codeflow?start=16&length=11)

possible to use python function? `rm-rf` is always dangerous.

Reply:
Should we use `bootloader.remove_image` ?

Reply:
We can't because remove_image need check bootloader info, but install failed at here doesn't update bootloader.

The bootloader will be update by line 612:
update_sonic_environment(bootloader, binary_image_version)

A partial installed image failed here only extract files to /host/image-version folder, so we only need to delete that folder.

Reply:
how about `shutil.rmtree` ?

Reply:
Fixed